aea/skills/base.py

```diff
@@ -627,30 +627,31 @@ def models(self) -> Dict[str, Model]:
 
     @classmethod
     def from_dir(
-        cls, directory: str, skill_context: Optional[SkillContext] = None
+        cls, directory: str, agent_context: Optional[AgentContext] = None
     ) -> "Skill":
         """
         Load the skill from a directory.
 
         :param directory: the directory to the skill package.
-        :param skill_context: the skill context
+        :param agent_context: the skill context
         :return: the skill object.
         """
         configuration = cast(
             SkillConfig,
             ComponentConfiguration.load(ComponentType.SKILL, Path(directory)),
         )
         configuration._directory = Path(directory)
-        return Skill.from_config(configuration, skill_context)
+        return Skill.from_config(configuration, agent_context)
 
     @classmethod
     def from_config(
-        cls, configuration: SkillConfig, skill_context: Optional[SkillContext] = None
+        cls, configuration: SkillConfig, agent_context: Optional[AgentContext] = None
```

Comment:
Why not make it required? What is the usecase stopping us from doing that?

Reply:
done